### PR TITLE
Create core-structs.h header file

### PR DESCRIPTION
**Pull request by Benned-H**
This branch introduces two new files:

1. `core-structs.h` - Header defining the structs we'll use throughout the project. When reviewing, please check for code standard deviations, as this is the first proper code we're reviewing.
2. `uncrustify.cfg` - Defines a default configuration for Uncrustify, an auto-formatter used on `core-structs.h`.

### CODE REVIEW

**Comment by Benned-H** [comment]
Should we put the Uncrustify config under a `tools` path? This can be easily changed.

**Comment by WillWu88** [comment]
> Should we put the Uncrustify config under a `tools` path? This can be easily changed.

If Uncrustify is just a config file like Doxygen, then I don't see why we can't put it in project root

File: uncrustify.cfg
Comment (WillWu88) [inline uncrustify.cfg:1]: Would it be too much work for you to summarize the important style checks into a document? This is pretty related to [ALP-65](https://willwu.atlassian.net/browse/ALP-65)

File: include/core-structs.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+/** A pose (position and orientation) in the 2D plane. */
+struct Pose2D {
+   float x;           // x-coordinate (scaled in meters)
+   float y;           // y-coordinate (scaled in meters)
+   float theta_rad;   // heading (radians) wrapped into [-pi, pi]
+};
+
+/**
+ * @brief A motion command for a 2D planar robot specified as egocentric velocity.
```
Comment (WillWu88) [inline include/core-structs.h:24]: It wouldn't be a bad idea to specify the frame in the docstrings

File: include/core-structs.h [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+/** An observation of a landmark as viewed from the robot's perspective. */
+struct Observation2D {
+   float range_m;     // Range from the robot to the landmark (meters)
+   float bearing_rad; // Bearing in the robot's frame to the landmark (radians)
+
+   /**
+    * @brief Stores the landmark signature corresponding to the observation.
+    *    Set to std::nullopt to represent an unknown correspondence variable.
+    *
+    * TODO: When DA is unknown, each particle's correspondence may differ.
```
Comment (WillWu88) [inline include/core-structs.h:40]: We can keep this `TODO` and investigate it later once we move to robot tests

**Review by WillWu88 — APPROVED** [review]
Looks good. I might have set up doxygen to only take in comments marked by `@brief`, so you may have to reformat some docstrings. Otherwise looks like a good place to start and we should get this in ASAP.

File: include/core-structs.h (hunk 2 above)
Comment (Benned-H) [reply include/core-structs.h:40]: Great, leaving as-is.

File: include/core-structs.h (hunk 1 above)
Comment (Benned-H) [reply include/core-structs.h:24]: Now changed.

**Review by WillWu88 — APPROVED** [review]

**Comment by WillWu88** [comment]
Re-approve to trigger unit-test after new file change

**Pull request merged**